src/main/java/com/agilethought/internship/university/service/impl/UniversityServiceImpl.java

```diff
@@ -75,4 +74,59 @@ else if(c.getCategory() == 3)
         return newList;
     }
 
+    public UpdateCourseResponse Updatecourse(UpdateCourseRequest request, String courseid) {
+
+        UpdateCourseResponse response = new UpdateCourseResponse();
+        Optional<Course> existcourse = repository.findById(courseid);
+
+        if (existcourse != null) {
+            Course newcourse = requestToUpdate(request, existcourse.get());
+            repository.save(newcourse);
+            if(newcourse.getCategory() == 1)
+                response.setCategory("JAVA");
+            else if(newcourse.getCategory() == 2)
+                response.setCategory("PEGA");
+            else if(newcourse.getCategory() == 3)
+                response.setCategory("JS");
+            response.setTitle(newcourse.getTitle());
+            response.setDescription(newcourse.getDescription());
+            response.setImg(newcourse.getImg());
+            response.setStatus(newcourse.getStatus());
+        }
+
+        return response;
+    }
+
+    private Course requestToUpdate(UpdateCourseRequest request, Course savecourse){
+        log.info("Category conversion successful");
```

Comment:
I think this log is not accurate. It says that "conversion was successful" but still lot of code is going to be executed. please move it a the end.